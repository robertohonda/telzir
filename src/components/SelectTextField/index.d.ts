import React from "react"
import { StandardProps } from "@material-ui/core"
import { TextFieldProps } from "@material-ui/core/TextField"
import { MenuItemProps } from "@material-ui/core/MenuItem"

export interface SelectTextFieldProps extends TextFieldProps {
  MenuItemProps: MenuItemProps
  options: Array<{ label: string; value: string }>
  cost: string
}

declare const CostCard: React.ComponentType<SelectTextFieldProps>

export default CostCard
