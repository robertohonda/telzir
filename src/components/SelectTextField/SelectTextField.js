import React from 'react'
import PropTypes from 'prop-types'
import TextField from '@material-ui/core/TextField'
import MenuItem from '@material-ui/core/MenuItem'

const SelectTextField = (props) => {
  const { options, MenuItemProps, ...textFieldProps } = props

  const renderOption = (option, index) => (
    <MenuItem key={index} value={option.value} {...MenuItemProps}>
      {option.label}
    </MenuItem>
  )

  return (
    <TextField select {...textFieldProps}>
      {options.map(renderOption)}
    </TextField>
  )
}

SelectTextField.propTypes = {
  options: PropTypes.arrayOf(
    PropTypes.shape({
      label: PropTypes.string.isRequired,
      value: PropTypes.string.isRequired,
    }).isRequired,
  ),
  MenuItemProps: PropTypes.object,
}

export default SelectTextField
