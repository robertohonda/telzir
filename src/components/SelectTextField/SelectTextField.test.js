import React from 'react'
import { shallow } from 'enzyme'
import SelectTextField from './SelectTextField'

describe('SelectTextField', () => {
  it('should render correctly', () => {
    const component = shallow(<SelectTextField options={[]} />)

    expect(component).toMatchSnapshot()
  })
})
