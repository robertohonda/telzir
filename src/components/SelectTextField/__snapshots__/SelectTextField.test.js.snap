// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectTextField should render correctly 1`] = `
<WithStyles(ForwardRef(TextField))
  select={true}
/>
`;
